DefunctRouteHandler/RemoveDefunctRouteHandler.php

```diff
@@ -42,7 +42,9 @@ public function handleDefunctRoutes(UriContextCollection $uriContextCollection)
             if (false === $uriContextCollection->containsAutoRoute($referringAutoRoute)) {
                 $newRoute = $uriContextCollection->getAutoRouteByTag($referringAutoRoute->getAutoRouteTag());
 
-                $this->adapter->migrateAutoRouteChildren($referringAutoRoute, $newRoute);
+                if (!empty($newRoute)) {
```

Comment:
If `getAutoRouteByTag` returns `null` then you should check for that: `if (null === $newRoute)`